### PR TITLE
ESP32 Factory Reset

### PR DESCRIPTION
**Pull request by unlogisch04**
Adding ESP32 factory reset. 
Adding Warning if hard coded credential are used.

### CODE REVIEW

File: src/serial/serialcommands.cpp [hunk 1]
```diff
@@ -78,9 +81,18 @@ namespace SerialCommands {
...
+                nvs_flash_erase();
         #else
-        // TODO: Implement eraseConfig for other boards
+            #warning SERIAL COMMAND FACTORY RESET NOT SUPPORTED
+            logger.info("FACTORY RESET NOT SUPPORTED");
+            return;
+        #endif
+        #endif
+        #if defined(WIFI_CREDS_SSID) && defined(WIFI_CREDS_PASSWD)
+            logger.error("FACTORY RESET can not clear your hardcoded WiFi crediential!");
```
Comment (TheDevMinerTV) [inline src/serial/serialcommands.cpp:95]:
```suggestion
            logger.warn("FACTORY RESET does not clear your hardcoded WiFi credentials!");
```

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: src/serial/serialcommands.cpp [hunk 2]
```diff
@@ -78,9 +81,18 @@ namespace SerialCommands {
 
         WiFi.disconnect(true); // Clear WiFi credentials
         #if ESP8266
-        ESP.eraseConfig(); // Clear ESP config
+            ESP.eraseConfig(); // Clear ESP config
+        #else 
+        #if ESP32
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp:86]: At the end i still need to delete the #endif

File: src/serial/serialcommands.cpp (hunk 1 above)
Comment (unlogisch04) [reply src/serial/serialcommands.cpp:95]: I did add also a Compiler warning.

File: src/serial/serialcommands.cpp [hunk 3]
```diff
@@ -78,9 +81,17 @@ namespace SerialCommands {
...
         #else
-        // TODO: Implement eraseConfig for other boards
+            #warning SERIAL COMMAND FACTORY RESET NOT SUPPORTED
+            logger.info("FACTORY RESET NOT SUPPORTED");
+            return;
+        #endif
+        #if defined(WIFI_CREDS_SSID) && defined(WIFI_CREDS_PASSWD)
+            #warning FACTORY RESET does not clear your hardcoded WiFi credentials!
+            logger.warn("FACTORY RESET does not clear your hardcoded WiFi credentials!");
         #endif
```
Comment (TheDevMinerTV) [inline src/serial/serialcommands.cpp:95]:
```suggestion
        #endif

```

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

**Pull request merged**